README.md

```diff
@@ -1,2 +1,214 @@
+
+As a baseline, Runtime uses the following commonly used tools that should already be installed on your machine (if you don't have these, ask the Runtime Project Manager to help you install them):
+
+* Python 3.7 or later: students write their student code in Python, so in order to run their code, we do need to use some Python in `executor`. To run Python code from C, we need to use Python's C API. Python's C API is notoriously unstable, and we wrote `executor` using Python 3.7 / 3.8.
+* `gcc`: short for "GNU C Compiler", we use this compiler to generate executables from C source code.
+
+Additionally, the Raspberry Pi uses a distribution of the Linux operating system (OS) called "Raspbian", a slight variant of the extremely popular "Debian" Linux. We tested Runtime to work well on Linux systems, but getting it to work properly on MacOS is difficult (and it definitely does not work on Windows). **It's highly recommended that you try to install a dual boot on your computer with some distribution of Linux (preferably Debian or Ubuntu) in order to properly build and run Runtime.**
+
+Runtime has the following third-party library dependencies:
+
+* `Cython`: this library is used by `executor` to implement the Student API in a way that is both callable from Runtime (which is written in C) and from student code (which is written in Python)
+    * Documentation: https://cython.readthedocs.io/en/latest/
+	* Installation (Debian/Raspbian Linux):
+	    1) `sudo apt-get -y install python3-distutils`   //get some utility functions
+        2) `sudo apt-get -y install python3-dev`         //get `<Python.h>`, `libpython3.7m.so`
+        3) `sudo apt-get -y install python3-pip`         //get `pip`
+        4) `python3 -m pip install Cython`               //get `Cython` using `pip`
+
+* Google `protobuf` and `protobuf-c`: Google `protobuf` is the library that we use to serialize our messages between Runtime and Shepherd, and Runtime and Dawn. The brief explanation of how it works is this: the user defines the structure of a message in "protobuf language", and saves it as a `.proto` file. Google's protobuf compiler will then take that `.proto` file and generate code that can be used in a desired target language to serialize and deserialize ("pack" and "unpack" in the language of protobufs) messages of that type. Since Google's protobuf compiler does not have native support for C, we need to use the third party library `protobuf-c` to generate C code. (But `protobuf-c` makes use of Google's library, so we still need it).
+    * `proto3` language documentation / guide: https://developers.google.com/protocol-buffers/docs/proto3
+	* `protobuf` Github: https://github.com/protocolbuffers/protobuf
+	* `protobuf-c` Github: https://github.com/protobuf-c/protobuf-c
+    * Installation (Debian/Raspbian Linux; MacOS):
+	    1) from `https://github.com/protocolbuffers/protobuf/releases`, download `protobuf-cpp-<release>.tar.gz` and extract it
```

Comment:
I would add the commands to download and extract. Ex: `wget https://github.com/protocolbuffers/protobuf/releases/download/v3.12.3/protobuf-cpp-3.12.3.tar.gz` then `tar xzf protobuf-cpp-3.12.3.tar.gz`.

Reply:
👌